presto-main/src/test/java/io/prestosql/server/security/oauth2/TestingHydraService.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server.security.oauth2;
+
+import io.prestosql.util.AutoCloseableCloser;
+import org.testcontainers.containers.FixedHostPortGenericContainer;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.startupcheck.OneShotStartupCheckStrategy;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.time.Duration;
+
+import static io.prestosql.util.RandomUtils.randomAlphanumeric;
+
+public class TestingHydraService
```

Comment:
There are a few alternatives such as: https://www.gluu.org/ and https://www.keycloak.org/ and a bunch of solutions built on top of Spring Framework Security lib such as: https://github.com/cloudfoundry/uaa. I've chosen ORY Hydra because it's lightweight, have well built docker image and extracts login & consent flow to an external app which might be handy in the future https://www.ory.sh/hydra/docs/login-consent-flow/.